src/decoder/vorbis.rs

```diff
@@ -44,12 +52,12 @@ impl<R> Source for VorbisDecoder<R> where R: Read + Seek {
 
     #[inline]
     fn get_channels(&self) -> u16 {
-        self.current_channels
+        self.stream_reader.ident_hdr.audio_channels as u16
```

Comment:
I thought that in the vorbis format, it was possible that the number of channels and samples rate changes between two frames.